api/v1/api.proto

```diff
@@ -0,0 +1,314 @@
+message PodFilter {
+        // If not empty, then the returned pods must have one of these names.
+        repeated string names = 1;
+
+        // If not empty, then the returned pods must be in one of these states.
+        repeated PodState states = 2;
+
+        // If not empty, then the returned pods must contain at least one of these images in the apps.
+        repeated string images = 3;
+
+        // If not empty, then the returned pods must be in at least one of these networks.
+        repeated string network_names = 4;
+}
+
+// ImageFilter defines the condition that the returned images need to satisfy in ListImages().
+// The conditions are combined by 'AND'.
+message ImageFilter {
+        // If not empty, then the returned images must have the listed ids.
+        repeated string ids = 1;
+
+        // if not empty, then the returned images must have one of the listed prefixes.
+        repeated string prefixes = 2;
+
+        // If not empty, then the returned images must contain one of the listed names as the base name.
+        // For example, both 'coreos.com/etcd' and 'k8s.io/etcd' will be returned if 'etcd' is included,
+        // however 'etcd-backup' will not be returned.
+        repeated string base_names = 3;
+
+        // If not empty, then the returned images' names must contain one of the listed key words.
+        // For example, both 'kubernetes-etcd', 'etcd:latest' will be returned if 'etcd' is included,
+        repeated string key_words = 4;
+
+        // If not empty, then the returned images must have one of the listed tags.
+        repeated string tags = 5;
+
+        // If set, then the returned images must be imported after this timestamp.
+        int64 imported_after = 6;
+
+        // If set, then the returned images must be imported before this timestamp.
+        int64 imported_before = 7;
+}
+
+// Info describes the information of rkt on the machine.
+message Info {
+        // Version of the rkt. In the form of Semantic Versioning (http://semver.org/).
+        string rkt_version = 1;
+
+        // Version of the appc. In the form of Semantic Versioning (http://semver.org/).
+        string appc_version = 2;
+}
+
+// EventType defines the type of the events that will be received via ListenEvents().
+enum EventType {
+        EVENT_TYPE_UNDEFINED = 0;
+
+        // Pod events.
+        EVENT_TYPE_POD_PREPARED          = 1;
+        EVENT_TYPE_POD_PREPARE_ABORTED   = 2;
+        EVENT_TYPE_POD_STARTED           = 3;
+        EVENT_TYPE_POD_EXITED            = 4;	// Also return reason or exit code??
+        EVENT_TYPE_POD_GARBAGE_COLLECTED = 5;
+
+        // Image events.
+        EVENT_TYPE_IMAGE_IMPORTED = 6;
+        EVENT_TYPE_IMAGE_REMOVED  = 7;
+}
+
+// Event describes the events that will be received via ListenEvents().
+message Event {
+        // Type of the event.
+        EventType type = 1;
+
+        // ID of the subject that causes the event, it can be the pod id or the image id.
```

Comment:
What is the plan to communicate individual container lifecycle events? This will be required for updating [ContainerStatus](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1080) in k8s API.